### DISCUSSION TITLE
[Task Groups] Previous tasks marked as failed if the current task fails in Grid UI

### DISCUSSION THREAD

**Discussion by rajatm91** [apache/airflow#25335] [Q&A] [answered]
Hi Everyone, 

We are using Task groups to group tasks in a within a dag. We have been observing a weird behaviour on the Grid UI for dags using task group where the task group previous run changes based on the current run status. 

For example, if the current scheduled run is in `running` state then previous state can also changes to `running`
![image-20220705-015400](https://user-images.githubusercontent.com/12770520/181222631-439eb4ca-ed6d-4ef7-b864-ed8b76afd9ea.png)

Similar if the current run has failed, then the previous run is also marked as failed. 

![image-20220705-015946](https://user-images.githubusercontent.com/12770520/181222804-c34db4b0-8bf4-4de6-937e-c8bdfd54007b.png)

**Airflow version: 2.3.1**

Any help on this would be really appreciated.

**Comment by potiuk** [comment, marked as answer]
Please upgrade to latest bugfix release 2.3.3 - It's been fixed there: https://github.com/apache/airflow/pull/24327

**Reply by rajatm91** [reply]
Thanks @potiuk, we upgrade and test it.
